default.nix

```diff
@@ -1,22 +1,33 @@
 let
-  config = {
-    packageOverrides = super: rec {
+  thisOverlay = self: super: {
+    my-R = self.rWrapper.override {
+      packages = with self.rPackages; [ dplyr ];
+    };
 
-      haskellPackages = super.haskellPackages.override {
-        overrides = haskellPackagesNew: haskellPackagesOld: {
-          nix-mwe = haskellPackagesNew.callPackage ./cabal.nix {
-            R = super.rWrapper.override {
-              packages = with super.rPackages; [ dplyr ];
-            };
-          };
-        };
-      };
+    haskellPackages = super.haskellPackages.override {
+      overrides = hself: hsuper: {
+        nix-mwe = self.haskell.lib.overrideCabal (hself.callCabal2nix "nix-mwe" ./. {}) (oldAttrs: {
+          buildTools = oldAttrs.buildTools or [] ++ [ self.makeWrapper ];
+          postInstall = oldAttrs.postInstall or "" + ''
+            wrapProgram $out/bin/nix-mwe --prefix 'PATH' ':' "${self.lib.getBin self.my-R}/bin"
+          '';
```

Comment:
I stole these lines from some other package in `nixpkgs/pkgs/development/haskell-modules/configuration-common.nix`.  

It appears that when you run `nix-mwe`, the `inline-r` package needs to load the R binary at runtime.  I guess it is running a repl behind the scenes.

I've tried to specify `my-R` (which contains `dplyr`), but it didn't seem to make a difference.